### PR TITLE
modifies so cellpose and stardist "work" on mac, also adds a StarDist2D

### PR DESCRIPTION
**Pull request by romainGuiet**

### CODE REVIEW

File: src/main/java/ch/epfl/biop/wrappers/cellpose/Cellpose.java [hunk 1]
```diff
@@ -62,48 +65,77 @@ public static void setVersion(String version) {
...
+
+            if (IJ.isWindows()) {
+                List<String> venv_activate_cmd = Arrays.asList("cmd.exe", "/C", new File(envDirPath, "Scripts/activate").toString());
+                cmd.addAll(venv_activate_cmd);
+            } else if ( IJ.isMacOSX()) {
+                throw new UnsupportedOperationException("Mac not supported yet");
+            }
+
         } else {
             System.out.println("Virtual env type unrecognized!");
```
Comment (NicoKiaru) [inline src/main/java/ch/epfl/biop/wrappers/cellpose/Cellpose.java:132]: Also throw new UnsupportedOperationException("Env type unrecognized");

File: src/main/java/ch/epfl/biop/wrappers/cellpose/Cellpose.java [hunk 2]
```diff
@@ -62,48 +65,77 @@ public static void setVersion(String version) {
...
 
-        // After starting the env we can now use cellpose
-        cmd.add("&");// to have a second line
-        List<String> cellpose_args_cmd = Arrays.asList("python", "-m", "cellpose");
-        cmd.addAll(cellpose_args_cmd);
-
-        // input options
-        cmd.addAll(options);
-
         System.out.println(cmd.toString().replace(",", ""));
```
Comment (NicoKiaru) [inline src/main/java/ch/epfl/biop/wrappers/cellpose/Cellpose.java:135]: Line to remove or not ? Useful for users '

File: src/main/java/ch/epfl/biop/wrappers/stardist/DefaultStardistTask.java [hunk 3]
```diff
@@ -18,11 +23,19 @@ public void run() throws Exception {
...
-            options.add("" + settings.z_tiles);
-            options.add("" + settings.y_tiles);
-            options.add("" + settings.x_tiles);
+        if(settings.dimension.equals(MODE3D)){
+            if ((settings.x_tiles > -1) && (settings.y_tiles > -1) && (settings.z_tiles > -1)) {
+                options.add("--n_tiles");
+                options.add("" + settings.z_tiles);
+                options.add("" + settings.y_tiles);
+                options.add("" + settings.x_tiles);
+            }
```
Comment (NicoKiaru) [inline src/main/java/ch/epfl/biop/wrappers/stardist/DefaultStardistTask.java:32]: What happens if you specify only x and y ? Should you support this ?

File: src/main/java/ch/epfl/biop/wrappers/stardist/Stardist.java [hunk 4]
```diff
@@ -36,11 +39,12 @@ public static void setStardistEnvType(String stardistEnvType) {
 
     static void execute(List<String> options, Consumer<InputStream> outputHandler) throws IOException, InterruptedException {
         List<String> cmd = new ArrayList<>();
+        List<String> start_cmd = null ;
 
         // Get the prefs about the env type
         String stardistEnvDirectory = Prefs.get(keyPrefix + "Stardist_envDirPath", Stardist.stardistEnvDirectory);
         String stardistEnvType = Prefs.get(keyPrefix + "Stardist_envType", Stardist.stardistEnvType);
-
+/*
```
Comment (NicoKiaru) [inline src/main/java/ch/epfl/biop/wrappers/stardist/Stardist.java:47]: Get rid of the commented code if it's not used anymore. We still have it in the git history if necessary anyway

File: src/main/java/ch/epfl/biop/wrappers/stardist/Stardist.java [hunk 5]
```diff
@@ -73,6 +77,72 @@ static void execute(List<String> options, Consumer<InputStream> outputHandler) t
...
+        }
+        cmd.addAll( start_cmd );
+
+
+        // Depending of the env type
+        if (stardistEnvType.equals("conda")) {
+            List<String> conda_activate_cmd = null;
+
+            if (IJ.isWindows()) {
+                // Activate the conda env
```
Comment (NicoKiaru) [inline src/main/java/ch/epfl/biop/wrappers/stardist/Stardist.java:97]: Did you check whether it's still working on windows ?

File: src/main/java/ch/epfl/biop/wrappers/stardist/Stardist.java [hunk 6]
```diff
@@ -73,6 +77,72 @@ static void execute(List<String> options, Consumer<InputStream> outputHandler) t
...
+
+            if (IJ.isWindows()) {
+                List<String> venv_activate_cmd = Arrays.asList("cmd.exe", "/C", new File(stardistEnvDirectory, "Scripts/activate").toString());
+                cmd.addAll(venv_activate_cmd);
+            } else if ( IJ.isMacOSX()) {
+                throw new UnsupportedOperationException("Mac not supported yet");
+            }
+
+        } else {
+            System.out.println("Virtual env type unrecognized!");
```
Comment (NicoKiaru) [inline src/main/java/ch/epfl/biop/wrappers/stardist/Stardist.java:139]: Same thing : throw new

File: src/main/java/ch/epfl/biop/wrappers/stardist/Stardist.java [hunk 7]
```diff
@@ -73,6 +77,72 @@ static void execute(List<String> options, Consumer<InputStream> outputHandler) t
...
+                    if (s.trim().contains(" "))
+                        return "\"" + s.trim() + "\"";
+                    return s;
+                }).collect(Collectors.toList());
+                // The last part needs to be sent as a single string, otherwise it does not run
+                String cmdString = cellpose_args_cmd.toString().replace(",","");
+
+                // finally add to cmd
+                cmd.add(cmdString.substring(1, cmdString.length()-1));
+            }
```
Comment (NicoKiaru) [inline src/main/java/ch/epfl/biop/wrappers/stardist/Stardist.java:127]: add else : throw new Unsupported etc. "OS unknown"

File: src/main/java/ch/epfl/biop/wrappers/stardist/StardistTaskSettings.java [hunk 8]
```diff
@@ -13,6 +16,11 @@ public class StardistTaskSettings {
     float prob_thresh = -1;
     float nms_thresh = -1;
 
+    public StardistTaskSettings setMode2D(){
```
Comment (NicoKiaru) [inline src/main/java/ch/epfl/biop/wrappers/stardist/StardistTaskSettings.java:19]: add a setMode3D as well

File: src/main/java/ch/epfl/biop/wrappers/stardist/ij2commands/StarDist2D_SegmentImgPath_Advanced.java [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+    @Parameter
+    int y_tiles = -1;
+
+    @Parameter(style = "format:#.0")
+    float min_norm = (float) 3.0;
+
+    @Parameter(style = "format:#.0")
+    float max_norm = (float) 99.8;
+
+    @Parameter
```
Comment (NicoKiaru) [inline src/main/java/ch/epfl/biop/wrappers/stardist/ij2commands/StarDist2D_SegmentImgPath_Advanced.java:36]: add a style as well "format:#.0", and a label : is is in percent ?

File: src/main/java/ch/epfl/biop/wrappers/stardist/ij2commands/StarDist2D_SegmentImgPath_Advanced.java [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+    float prob_thresh = -1;
+
+    @Parameter
```
Comment (NicoKiaru) [inline src/main/java/ch/epfl/biop/wrappers/stardist/ij2commands/StarDist2D_SegmentImgPath_Advanced.java:39]: Same thing : do you need to display significant digits ? then add a format tag in the style of the parameter

File: src/main/java/ch/epfl/biop/wrappers/stardist/ij2commands/StarDist2D_SegmentImgPlus_Advanced.java [hunk 11]
```diff
@@ -0,0 +1,135 @@
...
+    int y_tiles=-1;
+
+
+    @Parameter (style="format:#.0")
+    float min_norm= (float) 3.0;
+
+    @Parameter (style="format:#.0")
+    float max_norm = (float) 99.8;
+
+    @Parameter
```
Comment (NicoKiaru) [inline src/main/java/ch/epfl/biop/wrappers/stardist/ij2commands/StarDist2D_SegmentImgPlus_Advanced.java:43]: style ?

File: src/main/java/ch/epfl/biop/wrappers/stardist/ij2commands/StarDist2D_SegmentImgPlus_Advanced.java [hunk 12]
```diff
@@ -0,0 +1,135 @@
...
+    float prob_thresh = -1 ;
+
+    @Parameter
```
Comment (NicoKiaru) [inline src/main/java/ch/epfl/biop/wrappers/stardist/ij2commands/StarDist2D_SegmentImgPlus_Advanced.java:46]: style ?

**Review by NicoKiaru — COMMENTED** [review]
Here's a few comments, cheers.

File: src/main/java/ch/epfl/biop/wrappers/stardist/StardistTaskSettings.java (hunk 8 above)
Comment (romainGuiet) [reply src/main/java/ch/epfl/biop/wrappers/stardist/StardistTaskSettings.java:19]: done

File: src/main/java/ch/epfl/biop/wrappers/stardist/Stardist.java (hunk 5 above)
Comment (romainGuiet) [reply src/main/java/ch/epfl/biop/wrappers/stardist/Stardist.java:69]: yes

File: src/main/java/ch/epfl/biop/wrappers/stardist/Stardist.java (hunk 4 above)
Comment (romainGuiet) [reply src/main/java/ch/epfl/biop/wrappers/stardist/Stardist.java]: done

File: src/main/java/ch/epfl/biop/wrappers/cellpose/Cellpose.java [hunk 13]
```diff
@@ -62,48 +65,77 @@ public static void setVersion(String version) {
...
+                cellpose_args_cmd = cellpose_args_cmd.stream().map(s -> {
+                    if (s.trim().contains(" "))
+                        return "\"" + s.trim() + "\"";
+                    return s;
+                }).collect(Collectors.toList());
+                // The last part needs to be sent as a single string, otherwise it does not run
+                String cmdString = cellpose_args_cmd.toString().replace(",","");
+
+                // finally add to cmd
+                cmd.add(cmdString.substring(1, cmdString.length()-1));
```
[the remark this answers is not in the document]
Comment (romainGuiet) [reply src/main/java/ch/epfl/biop/wrappers/cellpose/Cellpose.java:119]: quote

**Pull request merged**